Phase one

RoyNyaga opened the pull request:
This branch of the project contains all the logic required for the battleship game to function as required. We created a ship class, board factory many others. We also tested different parts of the game to ensure coherency as development progresses.

RoyNyaga committed:
setup project

BenjaminGarza committed:
Created computer module, included generateRandom function

BenjaminGarza committed:
Created generateEducatedCoords in computer module

RoyNyaga committed:
created buildBoard function in domManipulation.js

RoyNyaga committed:
created initialSetup in domManipulations.js

BenjaminGarza committed:
Created gameBoard file and gameBoard class

BenjaminGarza committed:
Created playerAttack function and logic

BenjaminGarza committed:
Created computerAttack function

BenjaminGarza committed:
Created educated shot and random shot functions

to be used by computerAttack

BenjaminGarza committed:
Made receiveAttack function

RoyNyaga committed:
create info module and added checkForWin function in it

RoyNyaga committed:
added message function in info module

BenjaminGarza committed:
Added Ship factory function

BenjaminGarza committed:
Import and run initialSetup()

RoyNyaga committed:
added test for the computer factory

BenjaminGarza committed:
Add test for info and add test for ship

RoyNyaga committed:
described the project on readme file

RoyNyaga committed:
solved linting errors

RoyNyaga commented:
I have four linting errors I could not solve. These errors come from jest keywords specifically "expect" which eslint doesn't understand. I solved it at the level of individual test files but could not solve it at the level of the main.js file reason being that that file gets overridden each time npm start is run. Ideas on how to solve that will be very much welcome.
Again we tried implementing the drag and drop functionality but we could not figure out how to continue with it so we had to implement resetting which shuffles the position of the ships each time the reset button is pressed. So we decided to submit what we were able to do please consider our efforts. Thanks

shloch commented:
## project under review by SHEY LOUIS

shloch reviewed (changes requested):
## Hello @RoyNyaga 

Great job, works great
Nice alert messages 😊

## [ REQUIRED CHANGES ]
- change your `eslint.config` file to be [same as the one from MICROVERSE](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json)
- and your `.stickler.yml` file to correspond to [this one](https://github.com/microverseinc/linters-config/blob/master/javascript/.stickler.yml) 


The general configurations you'll be using for your js projects is [here](https://github.com/microverseinc/linters-config/tree/master/javascript)

- After this, you will get a whole lots of linting corrections to do

## [ OPTIONAL ]
- indicate on screen on top of the boards, which is User and which is computer
- keep a count on top of each of the boards of how many ships are sunk as we progressively play (or how many are left to be sunk)

Waiting for another re-review submission
Happy coding

RoyNyaga commented:
Thanks sir, will do the changes.


On Fri, Jan 24, 2020 at 10:41 PM SHEY Louis Chia <notifications@github.com>
wrote:

> *@shloch* requested changes on this pull request.
> Hello @RoyNyaga <https://github.com/RoyNyaga>
>
> Great job, works great
> Nice alert messages 😊
>
> [ REQUIRED CHANGES ]
>
>    - change your eslint.config file to be same as the one from MICROVERSE
>    <https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json>
>    - and your .stickler.yml file to correspond to this one
>    <https://github.com/microverseinc/linters-config/blob/master/javascript/.stickler.yml>
>
> The general configurations you'll be using for your js projects is here
> <https://github.com/microverseinc/linters-config/tree/master/javascript>
>
>    - After this, you will get a whole lots of linting corrections to do
>    [ OPTIONAL ]
>    - indicate on screen on top of the boards, which is User and which is
>    computer
>    - keep a count on top of each of the boards of how many ships are sunk
>    as we progressively play (or how many are left to be sunk)
>
> Waiting for another re-review submission
> Happy coding
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RoyNyaga/microverse-battleship-js/pull/1?email_source=notifications&email_token=ALZGFHICKOS2QPDDZVSHBM3Q7NOATA5CNFSM4KLJ5SFKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCTA3YOA#pullrequestreview-348240952>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALZGFHJLYM64QYHGX6UKSNLQ7NOATANCNFSM4KLJ5SFA>
> .
>

RoyNyaga committed:
added eslint configuration as stated by microverse requirement and fixed all errors

RoyNyaga committed:
ignored main.js from being scan by eslint

RoyNyaga committed:
fixing errors

RoyNyaga commented:
We Fixed all the required changes demanded by the last reviewer and solved all of the lint errors, We will continue with the optional changes after the project gets accepted just to save some time for ourselves. Thanks

jofer86 commented on eslint.config:
```diff
@@ -0,0 +1,9 @@
+    'no-param-reassign': 0,
+    'eol-last': 0,
+  }
+}
```
This file should be deleted, as we do not need two config files for our projects, this is the one stickler is listening to, and the one that is linting locally is fundamentally different, so we are bound to have errors, please do refer to [The Microverse Linters Repo](https://github.com/microverseinc/linters-config/tree/master/javascript); and follow the instructions step by step in order to be able to have a proper configuration. The stickler errors you might have with the configuration as it is now are not reflective of the errors you have on your local dev env. So let's make those changes.

jofer86 commented on .stickler.yml:
```diff
@@ -0,0 +1,17 @@
+    - 'testing/*'
+
+
+
```
you should not be ignoring `./estilnrc.json` as that is the one that should be listened to in line 3. Also this si fundamentally different from [This](https://github.com/microverseinc/linters-config/blob/master/javascript/.stickler.yml). Please refer to the comment on the eslint.config file.

jofer86 reviewed (changes requested):
## :wave: :wave: Hello!, Jorge here,  I'm here to review this Pull Request, let's take a look shall we :eye: :eye: :eye: 
- Changes marked as **[OPTIONAL]** are not hard requirements but nice to have.
- Below there are some required changes that need to be done in order for this project to be accepted.

--------------------------------------------------------------------
Cheers, 
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

RoyNyaga commented:
Ok sir will do the changes


On Mon, Jan 27, 2020 at 3:06 PM Jorge Fernando Rincón Figueroa <
notifications@github.com> wrote:

> *@jofer86* requested changes on this pull request.
> 👋 👋 Hello!, Jorge here, I'm here to review this Pull Request, let's
> take a look shall we 👁 👁 👁
>
>    - Changes marked as *[OPTIONAL]* are not hard requirements but nice to
>    have.
>    - Below there are some required changes that need to be done in order
>    for this project to be accepted.
>
> ------------------------------
>
> Cheers,
> Jorge Fernando Rincón Figueroa.
> Technical Support Engineer.
> *Slack: @jofer86 <https://github.com/jofer86>*
> ------------------------------
>
> In eslint.config
> <https://github.com/RoyNyaga/microverse-battleship-js/pull/1#discussion_r371257092>
> :
>
> > +{
>
> +  "parserOptions": { "ecmaVersion": 8 },
>
> +  "extends": "airbnb-base",
>
> +    rules: {
>
> +    'no-shadow': 0,
>
> +    'no-param-reassign': 0,
>
> +    'eol-last': 0,
>
> +  }
>
> +}
>
>
> This file should be deleted, as we do not need two config files for our
> projects, this is the one stickler is listening to, and the one that is
> linting locally is fundamentally different, so we are bound to have errors,
> please do refer to The Microverse Linters Repo
> <https://github.com/microverseinc/linters-config/tree/master/javascript>;
> and follow the instructions step by step in order to be able to have a
> proper configuration. The stickler errors you might have with the
> configuration as it is now are not reflective of the errors you have on
> your local dev env. So let's make those changes.
> ------------------------------
>
> In .stickler.yml
> <https://github.com/RoyNyaga/microverse-battleship-js/pull/1#discussion_r371257489>
> :
>
> > +linters:
>
> +    eslint:
>
> +        config: './eslint.config'
>
> +fixers:
>
> +    enable: true
>
> +
>
> +files:
>
> +  ignore:
>
> +    - 'bower_components/*'
>
> +    - 'node_modules/*'
>
> +    - '.eslintrc.json'
>
> +    - 'webpack.config.js'
>
> +    - 'dist/*'
>
> +    - 'testing/*'
>
> +
>
> +
>
> +
>
>
> you should not be ignoring ./estilnrc.json as that is the one that should
> be listened to in line 3. Also this si fundamentally different from This
> <https://github.com/microverseinc/linters-config/blob/master/javascript/.stickler.yml>.
> Please refer to the comment on the eslint.config file.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/RoyNyaga/microverse-battleship-js/pull/1?email_source=notifications&email_token=ALZGFHLTA25K5DWV5ICUPZLQ73S57A5CNFSM4KLJ5SFKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCTEIBTI#pullrequestreview-348684493>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALZGFHM56IYLIEH5JKMRRI3Q73S57ANCNFSM4KLJ5SFA>
> .
>

RoyNyaga committed:
removed eslintrc.json from linter ignore files and deleted eslint.config file

RoyNyaga committed:
adjusted eslint config file to be used

RoyNyaga committed:
changed .eslintrc.json to eslintrc.json

RoyNyaga commented:
I deleted the eslint.config file and removed the .eslintrc.json file from the list of eslint ignored files then fixed all the errors as demanded by the last reviewer. Thanks

shloch reviewed (approved):
# hello @RoyNyaga 

![accomplished](https://user-images.githubusercontent.com/40712459/73186937-dc5c0b00-4120-11ea-8b30-44ee1b29fd80.jpg)

## Approved !!

RoyNyaga commented:
Yuuuuppppppiiiiii thanks sir


On Mon, Jan 27, 2020 at 4:20 PM SHEY Louis Chia <notifications@github.com>
wrote:

> *@shloch* approved this pull request.
> hello @RoyNyaga <https://github.com/RoyNyaga>
>
> [image: accomplished]
> <https://user-images.githubusercontent.com/40712459/73186937-dc5c0b00-4120-11ea-8b30-44ee1b29fd80.jpg>
> Approved !!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RoyNyaga/microverse-battleship-js/pull/1?email_source=notifications&email_token=ALZGFHNJAD2PH75IXWF6PYLQ733TFA5CNFSM4KLJ5SFKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCTEWMPI#pullrequestreview-348743229>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALZGFHMP54TA5X4G5WD5EILQ733TFANCNFSM4KLJ5SFA>
> .
>
